io.openems.edge.battery.soltaro/src/io/openems/edge/battery/soltaro/single/versionb/SingleRack.java

```diff
@@ -393,8 +393,32 @@ private void writeValuesInApiChannels() {
 		Optional<Integer> minCellVoltageOpt = (Optional<Integer>) this
 				.channel(SingleRackChannelId.CLUSTER_1_MIN_CELL_VOLTAGE).value().asOptional();
 		if (minCellVoltageOpt.isPresent()) {
-			int voltageMillivolt = minCellVoltageOpt.get();
-			this.channel(Battery.ChannelId.MIN_CELL_VOLTAGE).setNextValue(voltageMillivolt);
+			int voltage_millivolt = minCellVoltageOpt.get();
+			this.channel(Battery.ChannelId.MIN_CELL_VOLTAGE).setNextValue(voltage_millivolt);
+		}
+		
+		@SuppressWarnings("unchecked")
+		Optional<Integer> maxCellVoltageOpt = (Optional<Integer>) this
+				.channel(SingleRackChannelId.CLUSTER_1_MAX_CELL_VOLTAGE).value().asOptional();
```

Comment:
Why don't you use direkt mapping in Modbus definition? If you really need "CLUSTER_1_MAX_CELL_VOLTAGE" also to have the same value, Modbus mapping also allows setting two channels from one register

Reply:
grown structures ;)

Reply:
i'll move it to define protocol